Src/FluentAssertions/Collections/StringCollectionAssertions.cs

```diff
@@ -190,5 +191,50 @@ public AndConstraint<StringCollectionAssertions> Contain(IEnumerable<string> exp
             args.AddRange(becauseArgs);
             return base.NotContain(unexpected, because, args.ToArray());
         }
+
+        /// <summary>
+        /// Asserts that the collection contains at least one string that matches a wildcard pattern.
+        /// </summary>
+        /// <param name="wildcardPattern">
+        /// The wildcard pattern with which the subject is matched, where * and ? have special meanings.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<StringCollectionAssertions> ContainMatch(string wildcardPattern, string because = "",
+            params object[] becauseArgs)
+        {
+            if (Subject is null)
+            {
+                Execute.Assertion
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith("Expected {context:collection} to contain a match of {0}{reason}, but found <null>.", wildcardPattern);
+
+                return new AndConstraint<StringCollectionAssertions>(this);
+            }
+
+            using (var scope = new AssertionScope())
```

Comment:
I just realized that this could be written as:
```c#
public AndConstraint<StringCollectionAssertions> ContainMatch(string wildcardPattern, string because = "",
    params object[] becauseArgs)
{
    Execute.Assertion
        .BecauseOf(because, becauseArgs)
        .ForCondition(Subject is object)
        .FailWith("Expected {context:collection} to contain a match of {0}{reason}, but found <null>.", wildcardPattern)
        .Then
        .ForCondition(ContainsMatch(wildcardPattern))
        .FailWith("Expected {context:collection} {0} to contain a match of {1}{reason}.", Subject, wildcardPattern);

    return new AndConstraint<StringCollectionAssertions>(this);
}

private bool ContainsMatch(string wildcardPattern)
{
    using (var scope = new AssertionScope())
    {
        foreach (string item in Subject)
        {
            item.Should().Match(wildcardPattern);
            bool isMatch = !scope.Discard().Any();
            if (isMatch)
            {
                return true;
            }
        }
    }

    return false;
}
```

That avoids the extra `return` statement in the bottom, which is effectively dead-code as it will never be hit.

Reply:
That's great, I was thinking of something similar with extracting the private method, but your version is more succinct.

Reply:
I just updated the example above to use `Any` instead in the `ContainsMatch` helping method.

Reply:
Even better.

If this gets any shorter, I'd have to rename the file to `StringCollectionAssertions.py` 😄